Azure ci flakes

Azure ci flakes out often and one of the errors I've seen is 

[pipeline1](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_image-builder/414/pull-azure-all/1316898623325212672#1:build-log.txt%3A377)
[pipeline2](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_image-builder/416/pull-azure-all/1317110189899386880#1:build-log.txt%3A456)

```
 An operation is currently performing on this storage account that requires exclusive access.
make: *** [Makefile:285: build-azure-vhd-ubuntu-1804] Error 1
make: *** Waiting for unfinished jobs.... 
```

Is it because ubuntu-1804 is getting built [here](https://github.com/kubernetes-sigs/image-builder/blob/05973968701f6da08a38769e5806f436add12cbc/images/capi/packer/azure/.pipelines/build-vhd.yaml#L23) & [here](https://github.com/kubernetes-sigs/image-builder/blob/05973968701f6da08a38769e5806f436add12cbc/images/capi/packer/azure/scripts/ci-azure-e2e.sh#L61) simultaneously?

CC: @CecileRobertMichon

--- a/images/capi/packer/azure/scripts/init-vhd.sh
+++ b/images/capi/packer/azure/scripts/init-vhd.sh
@@ -10,5 +10,7 @@ export RESOURCE_GROUP_NAME="${RESOURCE_GROUP_NAME:-cluster-api-images}"
 export AZURE_LOCATION="${AZURE_LOCATION:-southcentralus}"
 az group create -n ${RESOURCE_GROUP_NAME} -l ${AZURE_LOCATION} --tags ${TAGS}
 CREATE_TIME="$(date +%s)"
-export STORAGE_ACCOUNT_NAME="capi${CREATE_TIME}"
+RANDOM_SUFFIX="$(head /dev/urandom | LC_ALL=C tr -dc a-z | head -c 4 ; echo '')"
+export STORAGE_ACCOUNT_NAME="capi${CREATE_TIME}${RANDOM_SUFFIX}"
+az storage account check-name --name ${STORAGE_ACCOUNT_NAME}
 az storage account create -n ${STORAGE_ACCOUNT_NAME} -g ${RESOURCE_GROUP_NAME}

